SalaryAppTests/SalaryAppTests.swift

```diff
@@ -9,42 +9,110 @@ import XCTest
+        XCTAssertEqual(studentLoan,225)
 
-    func testWeeklySalaryLogic()  {
-        let weeklySalary = salaryVm.calculate(salary: salary, type: .weekly)
-        XCTAssertEqual(weeklySalary,138.46)
     }
     
-    func testDailySalaryLogic() {
-        let dailySalary = salaryVm.calculate(salary: salary, type: .daily)
-        XCTAssertEqual(dailySalary,1038.46)
+    func testPensionContributionCalculator() throws{
+        salary = 30000.0
+        let percentage = 20.0
+        let pensionContribution = pensionContributionCalculator.calculate(salary: salary, percentage: percentage)
+        XCTAssertEqual(pensionContribution,500)
+    }
+    
+    func testMonthlySalary() throws {
+    salary = 30000.0
+    let pensionContributions = 20.0
+    let salaryFormVM = SalaryFormViewModel(taxCalculator: TaxCalculator(), niCalculator: NICalaculator(), studentLoanCalculator: StudentLoanCalculator(), pensionContributionCalculator: PensionContributionCalculator())
         
+    let monthlyTotal = salaryFormVM.calculateMonthlySalary(salaryForm: SalaryForm(salary: salary, pensionContributions: pensionContributions))
+        XCTAssertEqual(monthlyTotal,1704.32)
     }
     
-    func testHourlySalaryLogic() {
-        let hourlySalary = salaryVm.calculate(salary: salary, type: .hourly)
-        XCTAssertEqual(hourlySalary,27.69)
+    func testDailySalary() throws {
+    salary = 30000.0
+    let pensionContributions = 20.0
+    let salaryFormVM = SalaryFormViewModel(taxCalculator: TaxCalculator(), niCalculator: NICalaculator(), studentLoanCalculator: StudentLoanCalculator(), pensionContributionCalculator: PensionContributionCalculator())
+        
+        let monthlyTotal = salaryFormVM.calculateDailySalary(salaryForm: SalaryForm(salary: salary, pensionContributions: pensionContributions))
+        XCTAssertEqual(monthlyTotal,372.6)
     }
     
-    override func tearDown() {
-        super.tearDown()
-        salaryVm = nil
-        salary = 0
+    func testHourlySalary() throws {
+    salary = 30000.0
+    let pensionContributions = 20.0
+    let salaryFormVM = SalaryFormViewModel(taxCalculator: TaxCalculator(), niCalculator: NICalaculator(), studentLoanCalculator: StudentLoanCalculator(), pensionContributionCalculator: PensionContributionCalculator())
+        
+        let monthlyTotal = salaryFormVM.calculateHourlySalary(salaryForm: SalaryForm(salary: salary, pensionContributions: pensionContributions))
+        XCTAssertEqual(monthlyTotal,571.47)
     }
     
-    override func setUp() {
-        super.setUp()
-        salaryVm = SalaryViewModel()
-        salary = 54000
+    func testWeelySalary() throws {
+    salary = 30000.0
+    let pensionContributions = 20.0
+    let salaryFormVM = SalaryFormViewModel(taxCalculator: TaxCalculator(), niCalculator: NICalaculator(), studentLoanCalculator: StudentLoanCalculator(), pensionContributionCalculator: PensionContributionCalculator())
```

Comment:
The salary form vm should be created in setup and destroyed in teardown do this throughout the whole project